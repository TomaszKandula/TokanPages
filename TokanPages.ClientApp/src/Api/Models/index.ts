export type { AddArticleDto } from "./Articles/addArticleDto";
export type { UpdateArticleContentDto } from "./Articles/updateArticleContentDto";
export type { UpdateArticleCountDto } from "./Articles/updateArticleCountDto";
export type { UpdateArticleLikesDto } from "./Articles/updateArticleLikesDto";
export type { UpdateArticleVisibilityDto } from "./Articles/updateArticleVisibilityDto";
export type { RemoveArticleDto } from "./Articles/removeArticleDto";

export type { SendMessageDto } from "./Mailer/sendMessageDto";
export type { SendNewsletterDto } from "./Mailer/sendNewsletterDto";
export type { SubscriberInfoDto } from "./Mailer/subscriberInfoDto";
export type { VerifyEmailAddressDto } from "./Mailer/verifyEmailAddressDto";

export type { AddSubscriberDto } from "./Subscribers/addSubscriberDto";
export type { UpdateSubscriberDto } from "./Subscribers/updateSubscriberDto";
export type { RemoveSubscriberDto } from "./Subscribers/removeSubscriberDto";

export type { UserDataDto } from "./Users/userDataDto";
export type { ActivateUserDto } from "./Users/activateUserDto";
export type { AddUserDto } from "./Users/addUserDto";
export type { UpdateUserDto } from "./Users/updateUserDto";
export type { RemoveUserDto } from "./Users/removeUserDto";
export type { AuthenticateUserDto } from "./Users/authenticateUserDto";
export type { AuthenticateUserResultDto } from "./Users/authenticateUserResultDto";
export type { ReAuthenticateUserDto } from "./Users/reAuthenticateUserDto";
export type { ReAuthenticateUserResultDto } from "./Users/reAuthenticateUserResultDto";
export type { RevokeUserRefreshTokenDto } from "./Users/revokeUserRefreshTokenDto";
export type { ResetUserPasswordDto } from "./Users/resetUserPasswordDto";
export type { UpdateUserPasswordDto } from "./Users/updateUserPasswordDto";
export type { UserPermissionDto } from "./Users/userPermissionDto";
export type { UserRoleDto } from "./Users/userRoleDto";
export type { UploadUserMediaDto } from "./Users/uploadUserMediaDto";
export type { UploadUserMediaResultDto } from "./Users/uploadUserMediaResultDto";

export type { ErrorDto } from "./Error/errorDto";
export type { ValidationErrorsDto } from "./Error/validationErrorsDto";

export type { DocumentContentDto } from "./Components/documentContentDto";

export type { NavigationContentDto } from "./Components/navigationContentDto";
export type { HeaderContentDto } from "./Components/headerContentDto";
export type { FooterContentDto } from "./Components/footerContentDto";
export type { ClientsContentDto } from "./Components/clientsContentDto";
export type { ArticleFeaturesContentDto } from "./Components/articleFeaturesContentDto";
export type { ContactFormContentDto } from "./Components/contactFormContentDto";
export type { CookiesPromptContentDto } from "./Components/cookiesPromptContentDto";
export type { FeaturedContentDto } from "./Components/featuredContentDto";
export type { FeaturesContentDto } from "./Components/featuresContentDto";
export type { NewsletterContentDto } from "./Components/newsletterContentDto";
export type { ResetPasswordContentDto } from "./Components/resetPasswordContentDto";
export type { UpdatePasswordContentDto } from "./Components/updatePasswordContentDto";
export type { UserSigninContentDto } from "./Components/userSigninContentDto";
export type { UserSignoutContentDto } from "./Components/userSignoutContentDto";
export type { UserSignupContentDto } from "./Components/userSignupContentDto";
export type { TestimonialsContentDto } from "./Components/testimonialsContentDto";
export type { UnsubscribeContentDto } from "./Components/unsubscribeContentDto";
export type { ActivateAccountContentDto } from "./Components/activateAccountContentDto";
export type { UpdateSubscriberContentDto } from "./Components/updateSubscriberContentDto";
export type { WrongPagePromptContentDto } from "./Components/wrongPagePromptContentDto";
export type { 
    AccountContentDto, 
    SectionAccessDenied,
    SectionAccountInformation,
    SectionAccountPassword,
    SectionAccountRemoval
} from "./Components/accountContentDto";

export type { IconDto } from "./Components/Common/iconDto";
export type { LinkDto } from "./Components/Common/linkDto";
export type { RowItemDto } from "./Components/Common/rowItemDto";
export type { TextItemDto } from "./Components/Common/textItemDto";
export type { ContentDto } from "./Components/Common/contentDto";

export type { GetContentManifestDto } from "./Content/getContentManifestDto";
export type { LanguageItemDto } from "./Content/Items/languageItemDto";
export type { JWT } from "./Jwt";
