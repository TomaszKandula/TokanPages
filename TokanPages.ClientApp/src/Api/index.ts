export {
    IsSuccessStatusCode,
    GetBaseHeaders,
    GetProcessedBody,
    GetProcessedHeaders,
    GetProcessedResponse,
} from "./Utilities";

export type { ExecuteApiActionResultProps, ExecuteStoreActionProps, ExecuteApiActionProps } from "./Utilities/Types";

export {
    API_BASE_URI,
    GET_ARTICLES,
    GET_ARTICLE,
    GET_ARTICLE_BY_TITLE,
    ADD_ARTICLE,
    UPDATE_ARTICLE_CONTENT,
    UPDATE_ARTICLE_COUNT,
    UPDATE_ARTICLE_LIKES,
    UPDATE_ARTICLE_VISIBILITY,
    RETRIEVE_ARTICLE_INFO,
    REMOVE_ARTICLE,
    LOG_MESSAGE,
    GET_USERS,
    GET_USER,
    GET_USER_IMAGE,
    ADD_USER,
    ACTIVATE_USER,
    UPDATE_USER,
    REMOVE_USER,
    AUTHENTICATE,
    REAUTHENTICATE,
    VERIFY_USER_EMAIL,
    REVOKE_REFRESH_TOKEN,
    REVOKE_USER_TOKEN,
    RESET_USER_PASSWORD,
    UPDATE_USER_PASSWORD,
    UPLOAD_USER_IMAGE,
    GET_NEWSLETTERS,
    GET_NEWSLETTER,
    ADD_NEWSLETTER,
    UPDATE_NEWSLETTER,
    REMOVE_NEWSLETTER,
    SEND_MESSAGE,
    SEND_NEWSLETTER,
    NOTIFY_WEB_URL,
    NOTIFICATION_STATUS,
    GET_CONTENT_MANIFEST,
    REQUEST_PAGE_DATA,
    GET_ARTICLE_MAIN_IMAGE_URL,
    GET_IMAGES_URL,
    GET_DOCUMENTS_URL,
    ARTICLE_PATH,
    GET_USER_VIDEO,
    GET_VIDEO_ASSET,
    GET_USER_NOTES,
    GET_USER_NOTE,
    ADD_USER_NOTE,
    UPDATE_USER_NOTE,
    REMOVE_USER_NOTE,
} from "./Paths";
