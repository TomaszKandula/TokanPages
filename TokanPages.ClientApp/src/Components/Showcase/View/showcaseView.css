.showcase-caption-text {
    text-align: center;
    font-size: 2rem !important;
    font-weight: 500 !important;
    color: #6367ef;
}

.showcase-feature-box {
    display: flex;
    height: 100%;
}

.showcase-feature-textbox {
    margin-top: auto;
    margin-bottom: auto;
}

.showcase-feature-text1 {
    font-size: 1.9rem !important;
    font-weight: 500 !important;
    margin: 0;
}

.showcase-feature-text2 {
    font-size: 1.125rem !important;
    font-weight: 400 !important;
    line-height: 1.8 !important;
    color: #757575 !important;
}

.showcase-feature-image {
    height: 512px;
}
