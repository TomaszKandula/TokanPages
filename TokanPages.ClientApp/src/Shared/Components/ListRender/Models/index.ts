export type { Fields } from "./Fields";
export type { Item } from "./Item";
export type { Subitem } from "./Subitem";
