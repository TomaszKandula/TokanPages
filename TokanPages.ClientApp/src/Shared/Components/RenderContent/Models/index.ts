export type { IArticleItem } from "./ArticleItemModel";
export type { IAuthor } from "./AuthorModel";
export type { ITextObject, ITextItem, IRowItem } from "./TextModel";
