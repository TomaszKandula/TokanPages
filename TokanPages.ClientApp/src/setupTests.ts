// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
