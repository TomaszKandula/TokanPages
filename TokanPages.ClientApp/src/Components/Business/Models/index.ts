import { ServiceItemDto, TechItemsDto } from "../../../Api/Models";
import { ReactChangeEvent, ReactMouseEvent } from "../../../Shared/types";

export interface MessageFormProps {
    company: string;
    firstName: string;
    lastName: string;
    email: string;
    phone: string;
    description: string;
    techStack: string[];
    services: string[];
}

export interface TechStackListProps {
    isDisabled: boolean;
    list: TechItemsDto[];
    handler: (event: ReactChangeEvent, isChecked: boolean) => void;
}

export interface ServiceItemCardProps {
    isDisabled: boolean;
    key: React.Key | null | undefined;
    value: ServiceItemDto;
    handler: (event: ReactMouseEvent) => void;
    services: string[];
}

export interface BusinessFormProps {
    hasCaption?: boolean;
    hasIcon?: boolean;
    hasShadow?: boolean;
    className?: string;
    background?: string;
}
