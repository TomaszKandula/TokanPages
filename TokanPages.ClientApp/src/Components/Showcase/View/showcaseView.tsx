import * as React from "react";
import { useSelector } from "react-redux";
import { Link } from "react-router-dom";
import { GET_SHOWCASE_IMAGE_URL } from "../../../Api";
import { FeatureShowcaseContentDto } from "../../../Api/Models";
import { ApplicationState } from "../../../Store/Configuration";
import { Animated, CustomImage, RenderHtml, Skeleton } from "../../../Shared/Components";
import Validate from "validate.js";
import "./showcaseView.css";

interface ShowcaseViewProps {
    className?: string;
}

interface ActiveButtonProps extends FeatureShowcaseContentDto {
    isLoading: boolean;
}

const ActiveButton = (props: ActiveButtonProps): React.ReactElement => {
    if (Validate.isEmpty(props?.action?.href)) {
        return <button className="bulma-button bulma-is-link bulma-is-light">{props?.action?.text}</button>;
    }

    return (
        <Link to={props?.action?.href ?? ""} className="link" rel="noopener nofollow">
            <button className="bulma-button bulma-is-link bulma-is-light">{props?.action?.text}</button>
        </Link>
    );
};

export const ShowcaseView = (props: ShowcaseViewProps): React.ReactElement => {
    const data = useSelector((state: ApplicationState) => state.contentPageData);
    const isLoading = data?.isLoading;
    const showcase = data?.components?.sectionShowcase;

    return (
        <section className={props.className}>
            <div className="bulma-container">
                <div className="py-6">
                    <Animated dataAos="fade-down">
                        <Skeleton isLoading={isLoading} mode="Text" height={40}>
                            <p className="is-size-3	has-text-centered has-text-link">
                                {showcase?.caption?.toUpperCase()}
                            </p>
                        </Skeleton>
                    </Animated>
                    <Animated dataAos="fade-up">
                        <div className="bulma-columns showcase-feature-columns">
                            <div className="bulma-column p-0 is-flex is-flex-direction-column is-align-self-center">
                                <Skeleton isLoading={isLoading} mode="Text">
                                    <h2 className="is-size-3 py-5 has-text-black">{showcase?.heading}</h2>
                                </Skeleton>
                                <Skeleton isLoading={isLoading} mode="Text">
                                    <RenderHtml
                                        value={showcase?.text}
                                        tag="p"
                                        className="is-size-5 py-3 has-text-grey line-height-18 showcase-feature-text"
                                    />
                                </Skeleton>
                                <Skeleton isLoading={isLoading} mode="Rect">
                                    <div className="has-text-left py-5">
                                        <ActiveButton isLoading={isLoading} {...showcase} />
                                    </div>
                                </Skeleton>
                            </div>
                            <div className="bulma-column p-0">
                                <div className="bulma-card">
                                    <div className="bulma-card-image">
                                        <figure className="bulma-image">
                                            <Skeleton isLoading={isLoading} mode="Rect" height={400}>
                                                <CustomImage
                                                    base={GET_SHOWCASE_IMAGE_URL}
                                                    source={showcase?.image}
                                                    className="showcase-feature-image"
                                                    title="Illustration"
                                                    alt="An image illustrating showcase section"
                                                />
                                            </Skeleton>
                                        </figure>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </Animated>
                </div>
            </div>
        </section>
    );
};
