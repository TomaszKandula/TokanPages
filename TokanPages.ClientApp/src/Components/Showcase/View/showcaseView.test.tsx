import "../../../setupTests";
import React from "react";
import * as Redux from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import { FeatureShowcaseContentDto } from "../../../Api/Models";
import { ContentPageData } from "../../../Store/Defaults";
import { ApplicationDefault, ApplicationState } from "../../../Store/Configuration";
import { ShowcaseView } from "./showcaseView";

jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn(),
}));

describe("test component: showcaseView", () => {
    const testContent: FeatureShowcaseContentDto = {
        language: "en",
        caption: "Showcase",
        heading: "Most notable projects",
        text: "Let's look at some of...",
        image: "image.webp",
        action: {
            text: "Go to showcase",
            href: "/en/showcase",
        },
    };

    let state: ApplicationState = ApplicationDefault;
    state.contentPageData = ContentPageData;
    state.contentPageData.components.sectionShowcase = testContent;

    const useSelectorMock = jest.spyOn(Redux, "useSelector");
    beforeEach(() => {
        useSelectorMock.mockImplementation(callback => callback(state));
    });

    it("should render correctly '<ShowcaseView />' when content is loaded.", () => {
        const html = render(
            <BrowserRouter>
                <ShowcaseView />
            </BrowserRouter>
        );

        expect(useSelectorMock).toBeCalledTimes(2);
        expect(html).toMatchSnapshot();
    });
});
