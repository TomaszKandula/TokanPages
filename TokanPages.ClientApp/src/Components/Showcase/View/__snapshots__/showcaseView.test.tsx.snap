// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component: showcaseView should render correctly '<ShowcaseView />' when content is loaded. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <div
          class="bulma-container"
        >
          <div
            class="py-6"
          >
            <div
              data-aos="fade-down"
              data-aos-delay="0"
            >
              <p
                class="is-size-3	has-text-centered has-text-link"
              >
                SHOWCASE
              </p>
            </div>
            <div
              data-aos="fade-up"
              data-aos-delay="0"
            >
              <div
                class="bulma-columns m-6"
              >
                <div
                  class="bulma-column p-0 is-flex is-flex-direction-column is-align-self-center"
                >
                  <h2
                    class="is-size-3 py-5 has-text-black"
                  >
                    Most notable projects
                  </h2>
                  <p
                    class="is-size-5 py-3 pr-6 has-text-grey line-height-18"
                  >
                    Let's look at some of...
                  </p>
                  <div
                    class="has-text-left py-5"
                  >
                    <a
                      class="link"
                      href="/en/showcase"
                      rel="noopener nofollow"
                    >
                      <button
                        class="bulma-button bulma-is-link bulma-is-light"
                      >
                        Go to showcase
                      </button>
                    </a>
                  </div>
                </div>
                <div
                  class="bulma-column p-0"
                >
                  <div
                    class="bulma-card"
                  >
                    <div
                      class="bulma-card-image"
                    >
                      <figure
                        class="bulma-image"
                      >
                        <img
                          alt="An image illustrating showcase section"
                          class="showcase-feature-image lazyloaded"
                          loading="lazy"
                          src="http://localhost:5000/api/v1/content/assets/getNonVideoAsset?blobName=images/sections/showcase/image.webp"
                          title="Illustration"
                        />
                      </figure>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <div
        class="bulma-container"
      >
        <div
          class="py-6"
        >
          <div
            data-aos="fade-down"
            data-aos-delay="0"
          >
            <p
              class="is-size-3	has-text-centered has-text-link"
            >
              SHOWCASE
            </p>
          </div>
          <div
            data-aos="fade-up"
            data-aos-delay="0"
          >
            <div
              class="bulma-columns m-6"
            >
              <div
                class="bulma-column p-0 is-flex is-flex-direction-column is-align-self-center"
              >
                <h2
                  class="is-size-3 py-5 has-text-black"
                >
                  Most notable projects
                </h2>
                <p
                  class="is-size-5 py-3 pr-6 has-text-grey line-height-18"
                >
                  Let's look at some of...
                </p>
                <div
                  class="has-text-left py-5"
                >
                  <a
                    class="link"
                    href="/en/showcase"
                    rel="noopener nofollow"
                  >
                    <button
                      class="bulma-button bulma-is-link bulma-is-light"
                    >
                      Go to showcase
                    </button>
                  </a>
                </div>
              </div>
              <div
                class="bulma-column p-0"
              >
                <div
                  class="bulma-card"
                >
                  <div
                    class="bulma-card-image"
                  >
                    <figure
                      class="bulma-image"
                    >
                      <img
                        alt="An image illustrating showcase section"
                        class="showcase-feature-image lazyloaded"
                        loading="lazy"
                        src="http://localhost:5000/api/v1/content/assets/getNonVideoAsset?blobName=images/sections/showcase/image.webp"
                        title="Illustration"
                      />
                    </figure>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
